stresstests/src/test/java/org/neo4j/kernel/stresstests/CheckPointingLogRotationStressTesting.java

```diff
@@ -0,0 +1,112 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.stresstests;
+
+import org.junit.Test;
+
+import java.io.File;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.factory.GraphDatabaseFactory;
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory;
+import org.neo4j.kernel.impl.logging.NullLogService;
+import org.neo4j.kernel.stresstests.tracers.TimerTransactionTracer;
+import org.neo4j.kernel.stresstests.workload.Workload;
+import org.neo4j.unsafe.impl.batchimport.ParallelBatchImporter;
+import org.neo4j.unsafe.impl.batchimport.staging.ExecutionMonitors;
+
+import static java.lang.Integer.parseInt;
+import static java.lang.Long.parseLong;
+import static java.lang.System.getProperty;
+import static java.lang.System.getenv;
+import static org.neo4j.kernel.stresstests.mutation.RandomMutationFactory.defaultRandomMutation;
+import static org.neo4j.unsafe.impl.batchimport.Configuration.DEFAULT;
+
+/**
+ * Notice the class name: this is _not_ going to be run as part of the main build.
+ */
+public class CheckPointingLogRotationStressTesting
+{
+    private static final String DEFAULT_DURATION_IN_MINUTES = "30";
+    private static final String DEFAULT_STORE_DIR = new File( getProperty( "java.io.tmpdir" ), "store" ).getPath();
+    private static final String DEFAULT_NODE_COUNT = "100000";
+    private static final String DEFAULT_WORKER_THREADS = "32";
+    private static final String DEFAULT_PAGE_CACHE_MEMORY = "4g";
+    private static final String DEFAULT_PAGE_SIZE = "8k";
+
+    @Test
+    public void shouldBehaveCorrectlyUnderStress() throws Throwable
+    {
+        long durationInMinutes =
+                parseLong( fromEnv( "CHECK_POINT_LOG_ROTATION_STRESS_DURATION", DEFAULT_DURATION_IN_MINUTES ) );
+        File storeDir = ensureExists( fromEnv( "CHECK_POINT_LOG_ROTATION_STORE_DIRECTORY", DEFAULT_STORE_DIR ) );
+        long nodeCount = parseLong( fromEnv( "CHECK_POINT_LOG_ROTATION_NODE_COUNT", DEFAULT_NODE_COUNT ) );
+        int threads = parseInt( fromEnv( "CHECK_POINT_LOG_ROTATION_WORKER_THREADS", DEFAULT_WORKER_THREADS ) );
+        String pageCacheMemory = fromEnv( "CHECK_POINT_LOG_ROTATION_PAGE_CACHE_MEMORY", DEFAULT_PAGE_CACHE_MEMORY );
+        String pageSize = fromEnv( "CHECK_POINT_LOG_ROTATION_PAGE_SIZE", DEFAULT_PAGE_SIZE );
+
+        System.out.println( "1/5  Building initial store..." );
```

Comment:
It is a stress test that will run separately from the main build so I think it is fine in having some printing. We can reconsider if this causes problems.